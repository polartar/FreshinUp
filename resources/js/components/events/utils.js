import get from 'lodash/get'

const getFileNameCopy = (name) => {
  const regex = /\s*\(([0-9]+)\)$/gm
  const matches = name.match(regex) || []
  const count = (
    parseInt(
      get(matches, '[0]', '')
        .replace('(', '')
        .replace(')', '')
    ) || 0
  ) + 1
  return `Copy of ${name.replace(get(matches, '[0]', ''), '')} (${count})`
}

export default getFileNameCopy
