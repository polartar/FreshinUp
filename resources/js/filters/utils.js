export function prefixStr (value, prefix = '@') {
  if (!value) return ''
  return value ? `${prefix} ${value}` : ''
}
