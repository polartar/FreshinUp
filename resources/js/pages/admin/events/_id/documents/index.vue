<template>
  <div>
    <v-layout
      row
      align-center
      pt-3
    >
      <v-btn
        flat
        small
        @click="backToEventDetails"
      >
        <div class="back-btn-inner">
          <v-icon>fas fa-arrow-left</v-icon>
          <span>Return to Events Details</span>
        </div>
      </v-btn>
    </v-layout>
    <v-layout
      row
      justify-space-between
      ma-2
    >
      <h2 class="white--text">
        {{ pageTitle }}
      </h2>
      <v-flex
        v-if="statuses.length"
        text-xs-right
        sm2
        xs12
      >
        <status-select
          :value="status"
          :options="statuses"
        />
      </v-flex>
    </v-layout>
    <v-divider />
    <br>
    <v-layout
      row
      wrap
      pa-2
      justify-space-between
      class="event-new-wrap"
    >
      <v-flex
        md9
        sm8
      >
        <v-tabs
          v-model="active"
          slider-color="primary"
        >
          <v-tab
            key="0"
            ripple
          >
            Event Document
          </v-tab>
          <v-tab
            key="1"
            ripple
          >
            Event Activity
          </v-tab>
          <v-tab-item
            key="0"
          >
            <document-section
              :statuses="documentStatuses"
              :documents="[]"
            />
          </v-tab-item>
          <v-tab-item
            key="1"
          >
            <message-list />
          </v-tab-item>
        </v-tabs>
      </v-flex>
      <v-flex
        md3
        sm4
        pl-3
      >
        <div
          class="mb-4"
        >
          <customer-summary
            :customer="customer"
            @onButtonClick="viewCustomerProfile"
          />
        </div>
        <div>
          <financial-summary
            :financial="financial"
            @onButtonClick="viewContact"
          />
        </div>
      </v-flex>
    </v-layout>
  </div>
</template>

<script>
import { mapGetters, mapActions } from 'vuex'
import StatusSelect from '~/components/events/StatusSelect'
import CustomerSummary from '~/components/events/CustomerSummary'
import FinancialSummary from '~/components/events/FinancialSummary'
import DocumentSection from '~/components/events/DocumentSection'
import MessageList from '~/components/events/MessageList'

export default {
  layout: 'admin',
  components: {
    StatusSelect,
    CustomerSummary,
    FinancialSummary,
    DocumentSection,
    MessageList
  },
  data () {
    return {
      financial: {},
      active: 0
    }
  },
  computed: {
    ...mapGetters('page', ['isLoading']),
    ...mapGetters('events', { event: 'item' }),
    ...mapGetters('eventStatuses', { 'statuses': 'items' }),
    ...mapGetters('documentStatuses', { 'documentStatuses': 'items' }),
    pageTitle () {
      return this.event && this.event.name
    },
    status () {
      return this.event && this.event.host_status
    },
    customer () {
      return this.event ? {
        uuid: this.event.uuid,
        status: this.event.host_status,
        updated_at: this.event.updated_at,
        created_at: this.event.created_at
      } : {}
    }
  },
  methods: {
    ...mapActions('page', {
      setPageLoading: 'setLoading'
    }),
    backToEventDetails () {
      this.$router.push({ path: '/admin/events/' + this.event.uuid + '/edit' })
    },
    viewCustomerProfile () {
      if (this.event && this.event.host) {
        this.$router.push({ path: '/admin/companies/' + this.event.host.id })
      }
    },
    viewContact () {
    }
  },
  beforeRouteEnterOrUpdate (vm, to, from, next) {
    vm.setPageLoading(true)
    const eventUuid = to.path.split('/')[3]

    let eventParams = { id: eventUuid }

    Promise.all([
      vm.$store.dispatch('events/getItem', { params: eventParams }),
      vm.$store.dispatch('eventStatuses/getItems'),
      vm.$store.dispatch('documentStatuses/getItems')
    ]).then(() => {
      vm.$store.dispatch('page/setLoading', false)
      if (next) next()
    })
  }
}
</script>

<style scoped>
.back-btn-inner{
  color: #fff;
  display: flex;
  align-items: center;
  font-size: 13px;
}
.back-btn-inner span{
  margin-left: 10px;
  font-weight: bold;
  text-transform: initial;
}
.back-btn-inner .v-icon{
  font-size: 16px;
}
.service-summary{
  margin-top: 15px;
}
</style>
