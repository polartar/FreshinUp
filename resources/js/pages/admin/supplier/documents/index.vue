<script>
import Page from '~/pages/admin/docs/index.vue'
import { mapGetters } from 'vuex'
export default {
  extends: Page,
  computed: {
    ...mapGetters(['currentUser']),
    ...mapGetters('suppliers/documents', {
      docs: 'items',
      pagination: 'pagination',
      sorting: 'sorting',
      sortBy: 'sortBy',
      sortables: 'sortables'
    })
  },
  beforeRouteEnterOrUpdate (vm, to, from, next) {
    vm.$store.dispatch('suppliers/documents/setFilters', {
      ...vm.$route.query
    })
    vm.setPageLoading(true)
    Promise.all([
      vm.$store.dispatch('suppliers/documents/getItems', {
        params: {
          supplierId: vm.currentUser.uuid
        }
      }),
      vm.$store.dispatch('documentStatuses/getItems'),
      vm.$store.dispatch('documentTypes/getItems')
    ])
      .then(() => {
        if (next) next()
      })
      .catch(error => console.error(error))
      .then(() => vm.setPageLoading(false))
  }
}
</script>
