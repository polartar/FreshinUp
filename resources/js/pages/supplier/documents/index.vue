
<script>
import { mapGetters } from 'vuex'
import Documents from '~/pages/admin/docs/index.vue'

export default {
  extends: Documents,
  computed: {
    ...mapGetters(['currentUser'])
  },
  beforeRouteEnterOrUpdate (vm, to, from, next) {
    vm.$store.dispatch('documents/setFilters', {
      ...vm.$route.query,
      assigned_uuid: vm.currentUser.uuid
    })
    vm.setPageLoading(true)
    Promise.all([
      vm.$store.dispatch('documentStatuses/getItems'),
      vm.$store.dispatch('documentTypes/getItems')
    ])
      .then(() => {
        if (next) next()
      })
      .catch(error => console.error(error))
      .then(() => vm.setPageLoading(false))
  }
}
</script>
