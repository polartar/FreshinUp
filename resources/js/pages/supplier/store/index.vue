
<script>
import stores from '~/pages/admin/fleet-members/index.vue'
import { mapGetters } from 'vuex'
const INCLUDE = [
  'tags',
  'addresses',
  'status',
  'owner',
  'type'
]

export default {
  extends: stores,
  computed: {
    ...mapGetters(['currentUser'])
  },
  beforeRouteEnterOrUpdate (vm, to, from, next) {
    vm.setPageLoading(true)
    Promise.all([
      vm.$store.dispatch('stores/setFilters', {
        ...vm.$route.query,
        ...this.lastFilterParams,
        owner_uuid: vm.currentUser.uuid
      }),
      vm.$store.dispatch('storeStatuses/getItems', {
        params: { include: INCLUDE }
      })
    ])
      .then(() => {
        if (next) next()
      })
      .catch((error) => console.error(error))
      .then(() => vm.setPageLoading(false))
  }
}
</script>
