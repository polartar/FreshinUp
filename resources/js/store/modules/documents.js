import { makeModule } from '../utils/makeRestStore'
import { buildApi } from '@freshinup/core-ui/src/store/utils/makeRestStore'
export default (initialState = {}) => {
  const { items, item } = initialState
  const documentsApi = buildApi('foodfleet/documents', { items, item })
  const store = makeModule(documentsApi.getStore(), 'documents')

  const sortables = [
    { value: '-created_at', text: 'Newest' },
    { value: 'created_at', text: 'Oldest' },
    { value: 'title', text: 'Title (A - Z)' },
    { value: '-title', text: 'Title (Z - A)' }
  ]

  // Initial State
  store.state = {
    ...store.state,
    sortables
  }
  // Add Mutations
  store.mutations = {
    ...store.mutations,
    sortBy (state, value) {
      state.sortBy = value
    }
  }
  // Add Actions
  store.actions = {
    ...store.actions,
    sortBy ({ commit, dispatch }, value) {
      commit('UPDATE_SORT', { sortBy: value })
    }
  }

  // Add Getters
  store.getters = {
    ...store.getters
  }

  return {
    namespaced: true,
    ...store
  }
}
