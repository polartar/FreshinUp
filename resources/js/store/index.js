import { createStore } from 'fresh-bus/store'

import page from './modules/page'
import devices from './modules/devices'
import financialModifiers from './modules/financialModifiers'
import financialReports from './modules/financialReports'
import paymentTypes from './modules/paymentTypes'
import financialsummary from './modules/financialsummary'
import squares from './modules/squares'
import documents from './modules/documents'
import documentStatuses from './modules/documentStatuses'
import documentTypes from './modules/documentTypes'
import transactions from './modules/transactions'
import companyDetails from './modules/companyDetails'
import eventSummary from './modules/eventSummary'
import eventTypes from './modules/eventTypes'
import eventHistories from './modules/eventHistories'
import venues from './modules/venues'
import stores from './modules/stores'
import storeStatuses from './modules/storeStatuses'
import storeTypes from './modules/storeTypes'
import companies from './modules/companies'

export default (initialState = {}) => {
  return createStore(
    initialState,
    {
      modules: {
        page: page({}),
        devices: devices({}),
        financialModifiers: financialModifiers({}),
        financialReports: financialReports({}),
        paymentTypes: paymentTypes({}),
        financialsummary: financialsummary({}),
        squares: squares({}),
        documents: documents({}),
        documentStatuses: documentStatuses({}),
        documentTypes: documentTypes({}),
        transactions: transactions({}),
        companyDetails: companyDetails({}),
        eventSummary: eventSummary({}),
        eventTypes: eventTypes({}),
        eventHistories: eventHistories({}),
        venues: venues({}),
        stores: stores({}),
        storeStatuses: storeStatuses({}),
        storeTypes: storeTypes({}),
        companies: companies({})
      }
    }
  )
}
