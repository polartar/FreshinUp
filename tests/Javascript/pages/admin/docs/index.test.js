import { shallowMount } from '@vue/test-utils'
import { createStore } from 'fresh-bus/store'
import { createLocalVue } from 'fresh-bus/tests/utils'
import { FIXTURE_DOCUMENTS_RESPONSE, FIXTURE_DOCUMENTS_SORTED_BY_FIRSTNAME } from 'tests/__data__/documents'
import Component from '~/pages/admin/docs/index.vue'
import documents from '~/store/modules/documents'
import documentStatuses from '~/store/modules/documentStatuses'
import documentTypes from '~/store/modules/documentTypes'

describe('Admin Docs Page', () => {
  let localVue, mock, store, actions
  describe('Mount', () => {
    beforeEach(() => {
      const vue = createLocalVue({ validation: true })
      localVue = vue.localVue
      mock = vue.mock
    })

    afterEach(() => {
      mock.restore()
    })

    test('snapshot', async () => {
      const vue = createLocalVue({ validation: true })
      localVue = vue.localVue
      mock = vue.mock
        .onGet('api/foodfleet/documents', { params: { 'page[size]': 10, 'page[number]': 1, sort: 'first_name' } })
        .reply(200, { data: FIXTURE_DOCUMENTS_SORTED_BY_FIRSTNAME })
        .onGet('api/foodfleet/documents').reply(200, FIXTURE_DOCUMENTS_RESPONSE)
        .onAny()
        .reply(config => {
          console.warn('No mock match for ' + config.url, config)
          return [404, {}]
        })
      const store = createStore({}, {
        modules: {
          documents: documents({}),
          documentStatuses: documentStatuses({}),
          documentTypes: documentTypes({})
        }
      })
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        store
      })
      // Action: change State Machine's state
      await wrapper.vm.$store.dispatch('page/setLoading', false)
      await wrapper.vm.$nextTick()
      await wrapper.vm.$store.dispatch('documents/getItems')
      await wrapper.vm.$nextTick()
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Methods', () => {
    beforeEach(() => {
      const vue = createLocalVue({ validation: true })
      const docModule = documents({})
      localVue = vue.localVue
      actions = {
        patchItem: jest.fn()
      }
      store = createStore({
        docs: {
          items: FIXTURE_DOCUMENTS_RESPONSE
        }
      }, {
        modules: {
          documents: { ...docModule, actions: { ...docModule.actions, ...actions } }
        }
      })
    })

    afterEach(() => {
      mock.restore()
    })

    test('changeStatus function change doc status', async () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        store
      })

      wrapper.vm.changeStatus(2, { uuid: 'mock uuid' })
      const data = { data: { status: 2 }, params: { id: 'mock uuid' } }
      expect(actions.patchItem).toHaveBeenCalled()
      expect(actions.patchItem.mock.calls).toHaveLength(1)
      expect(actions.patchItem.mock.calls[0][1]).toEqual(data)
    })

    test('changeStatusMultiple function change doc status for each', async () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        store
      })

      wrapper.vm.changeStatusMultiple(3, [{ uuid: 'mock uuid 1' }, { uuid: 'mock uuid 2' }])

      expect(actions.patchItem).toHaveBeenCalled()
      expect(actions.patchItem.mock.calls).toHaveLength(2)

      const firstData = { data: { status: 3 }, params: { id: 'mock uuid 1' } }
      const secondData = { data: { status: 3 }, params: { id: 'mock uuid 2' } }

      expect(actions.patchItem.mock.calls[0][1]).toEqual(firstData)
      expect(actions.patchItem.mock.calls[1][1]).toEqual(secondData)
    })

    test('onPaginate function change paginate', () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        store
      })

      wrapper.vm.onPaginate({
        rowsPerPage: 2,
        totalItems: 5,
        page: 2
      })
      expect(wrapper.vm.pagination.rowsPerPage).toBe(2)
    })

    test('deleteDoc function change deleteTemp', () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        store
      })

      wrapper.vm.deleteDoc({ id: 1 })
      expect(wrapper.vm.deleteTemp[0].id).toBe(1)
      expect(wrapper.vm.deleteDialog).toBeTruthy()

      wrapper.vm.deleteDoc([{ id: 1 }])
      expect(wrapper.vm.deleteTemp[0].id).toBe(1)
      expect(wrapper.vm.deleteDialog).toBeTruthy()
    })
  })
})
