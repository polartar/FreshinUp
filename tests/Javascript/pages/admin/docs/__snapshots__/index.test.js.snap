// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Docs Page Mount snapshot 1`] = `
<div>
  <v-flex-stub
    align-center=""
    d-flex=""
    justify-space-between=""
    ma-2=""
    tag="div"
  >
    <h2
      class="white--text"
    >
      
      Document List
    
    </h2>
     
    <v-flex-stub
      tag="div"
      text-xs-right=""
    >
      <v-btn-stub
        activeclass="v-btn--active"
        color="primary"
        dark="true"
        slot="activator"
        tag="button"
        type="button"
      >
        
        Add New Content
      
      </v-btn-stub>
    </v-flex-stub>
  </v-flex-stub>
   
  <v-divider-stub />
   
  <br />
   
  <doc-filter-stub
    autocompleteurl=""
    sortables="[object Object],[object Object],[object Object],[object Object]"
    statuses=""
    types=""
  />
   
  <doctable-list-stub
    docs=""
    page="1"
    rowsperpage="10"
    rowsperpageitems="5,10,15,25,30,50"
    sortby=""
    statuses=""
    totalitems="50"
  />
   
  <v-dialog-stub
    contentclass=""
    maxwidth="500"
    origin="center center"
    transition="dialog-transition"
    width="auto"
  >
    <simple-confirm-stub
      cancellabel="No"
      class=""
      oklabel="Yes"
      title="Are you sure you want to delete this document?"
    >
      <div
        class="py-5 px-2"
      >
        <p
          class="subheading"
        >
          <span>
            Document
          </span>
          
            : 
          
        </p>
      </div>
    </simple-confirm-stub>
  </v-dialog-stub>
</div>
`;
